### PR TITLE
Update CSI Spec

### PR DESCRIPTION
**Pull request by yati1998**
this commit updates csi spec version to
include latest changes for volume group
snapshot

```release-note
Update CSI spec to a commit after v1.10.0 where VolumeGroupSnapshot moved to GA
```

### CODE REVIEW

**Comment by xing-yang** [comment]
We need to update again after CSI Spec is officially released.

**Comment by xing-yang** [comment]
Please add a release note.

**Comment by yati1998** [comment]
> We need to update again after CSI Spec is officially released.

sure, will add a TODO for that. lets get it for now.

File: go.mod [hunk 1]
```diff
@@ -3,7 +3,8 @@ module github.com/kubernetes-csi/external-snapshotter/v8
 go 1.23.1
 
 require (
-	github.com/container-storage-interface/spec v1.9.0
+	// TODO: update version once it is officially released
+	github.com/container-storage-interface/spec v1.10.1-0.20241022120259-f6b6d53db606
```
Comment (xing-yang) [inline go.mod:7]: Update csi-test https://github.com/kubernetes-csi/csi-test/releases/tag/v5.3.1

**Comment by xing-yang** [comment]
You probably need to take a look at this PR: https://github.com/kubernetes-csi/external-provisioner/pull/1262

**Comment by xing-yang** [comment]
Make this change, then it should work:
```
$ git diff main_test.go
diff --git a/cmd/csi-snapshotter/main_test.go b/cmd/csi-snapshotter/main_test.go
index c3ecf9d5..05c93e9e 100644
--- a/cmd/csi-snapshotter/main_test.go
+++ b/cmd/csi-snapshotter/main_test.go
@@ -23,6 +23,7 @@ import (

        "github.com/container-storage-interface/spec/lib/go/csi"
        "github.com/golang/mock/gomock"
+       "github.com/kubernetes-csi/csi-test/v5/utils"
        "github.com/kubernetes-csi/csi-lib-utils/connection"
        "github.com/kubernetes-csi/csi-lib-utils/metrics"
        "github.com/kubernetes-csi/csi-test/v5/driver"
@@ -125,7 +126,7 @@ func Test_supportsControllerCreateSnapshot(t *testing.T) {
                }

                // Setup expectation
-               controllerServer.EXPECT().ControllerGetCapabilities(gomock.Any(), in).Return(out, injectedErr).Times(1)
+               controllerServer.EXPECT().ControllerGetCapabilities(gomock.Any(), utils.Protobuf(in)).Return(out, injectedErr).Times(1)
```

**Comment by xing-yang** [comment]
Also here:
```
$ git diff snapshotter_test.go
diff --git a/pkg/snapshotter/snapshotter_test.go b/pkg/snapshotter/snapshotter_test.go
index 148848ff..8d6c79ef 100644
--- a/pkg/snapshotter/snapshotter_test.go
+++ b/pkg/snapshotter/snapshotter_test.go
@@ -25,6 +25,7 @@ import (

        "github.com/container-storage-interface/spec/lib/go/csi"
        "github.com/golang/mock/gomock"
+       "github.com/kubernetes-csi/csi-test/v5/utils"
        "github.com/kubernetes-csi/csi-lib-utils/connection"
        "github.com/kubernetes-csi/csi-lib-utils/metrics"
        "github.com/kubernetes-csi/csi-test/v5/driver"
@@ -210,7 +211,7 @@ func TestCreateSnapshot(t *testing.T) {
                // Setup expectation
                if in != nil {
                        identityServer.EXPECT().GetPluginInfo(gomock.Any(), gomock.Any()).Return(pluginInfoOutput, nil).Times(1)
-                       controllerServer.EXPECT().CreateSnapshot(gomock.Any(), in).Return(out, injectedErr).Times(1)
+                       controllerServer.EXPECT().CreateSnapshot(gomock.Any(), utils.Protobuf(in)).Return(out, injectedErr).Times(1)
                }

                s := NewSnapshotter(csiConn)
@@ -318,7 +319,7 @@ func TestDeleteSnapshot(t *testing.T) {

                // Setup expectation
                if in != nil {
-                       controllerServer.EXPECT().DeleteSnapshot(gomock.Any(), in).Return(out, injectedErr).Times(1)
+                       controllerServer.EXPECT().DeleteSnapshot(gomock.Any(), utils.Protobuf(in)).Return(out, injectedErr).Times(1)
                }

                s := NewSnapshotter(csiConn)
@@ -463,7 +464,7 @@ func TestGetSnapshotStatus(t *testing.T) {
                                Capabilities: controllerCapabilities,
                        }, nil).Times(1)
                        if test.listSnapshotsSupported {
-                               controllerServer.EXPECT().ListSnapshots(gomock.Any(), in).Return(out, injectedErr).Times(1)
+                               controllerServer.EXPECT().ListSnapshots(gomock.Any(), utils.Protobuf(in)).Return(out, injectedErr).Times(1)
                        }
                }
```

**Comment by xing-yang** [comment]
/lgtm
/approve

Trivy failure is unrelated to this change.

**Pull request merged**